templates/distribution/scripts/pre-apply.sh.tpl

```diff
@@ -9,29 +9,79 @@ vendorPath="{{ .paths.vendorPath }}"
-  $kustomizebin build $vendorPath/modules/logging/katalog/cerebro | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
-  echo "Waiting 3 minutes"
-  sleep 180
-  $kubectlbin delete -l app.kubernetes.io/name=opensearch pvc -n logging --wait --timeout=180s
+
+  $kustomizebin build $vendorPath/modules/logging/katalog/opensearch-dashboards > delete-opensearch-cerebro.yaml
+  $kustomizebin build $vendorPath/modules/logging/katalog/opensearch-triple >> delete-opensearch-cerebro.yaml
+  $kustomizebin build $vendorPath/modules/logging/katalog/cerebro >> delete-opensearch-cerebro.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-opensearch-cerebro.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-opensearch-cerebro-filtered.yaml
+    cp delete-opensearch-cerebro-filtered.yaml delete-opensearch-cerebro.yaml
+  fi
+{{- end }}
+
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-opensearch-cerebro.yaml
+  $kubectlbin delete --ignore-not-found -l app.kubernetes.io/name=opensearch pvc -n logging --wait --timeout=180s
   echo "Opensearch resources deleted"
 }
 
 deleteLoki() {
-  $kustomizebin build $vendorPath/modules/logging/katalog/loki-distributed | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
-  echo "Waiting 3 minutes"
-  sleep 180
-  $kubectlbin delete -l app.kubernetes.io/name=loki-distributed pvc -n logging --wait --timeout=180s
+
+  $kustomizebin build $vendorPath/modules/logging/katalog/loki-distributed > delete-loki.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-loki.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-loki-filtered.yaml
+    cp delete-loki-filtered.yaml delete-loki.yaml
+  fi
+{{- end }}
+
+  $kubectlbin delete --ignore-not-found --wait --timeout=180s -f delete-loki.yaml
+  $kubectlbin delete --ignore-not-found -l app.kubernetes.io/name=loki-distributed pvc -n logging --wait --timeout=180s
   echo "Loki resources deleted"
 }
 
+deleteLoggingOperator() {
+
+  $kustomizebin build $vendorPath/modules/logging/katalog/logging-operated | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+  $kustomizebin build $vendorPath/modules/logging/katalog/configs | $kubectlbin delete --ignore-not-found --wait --timeout=180s -f -
+
+  $kustomizebin build $vendorPath/modules/logging/katalog/minio-ha > delete-logging-minio-ha.yaml
+
+{{- if eq .spec.distribution.modules.monitoring.type "none" }}
+  if ! $kubectlbin get apiservice v1.monitoring.coreos.com; then
+    cat delete-logging-minio-ha.yaml | $yqbin 'select(.apiVersion != "monitoring.coreos.com/v1")' > delete-logging-minio-ha-filtered.yaml
+    cp delete-logging-minio-ha-filtered.yaml delete-tracing-minio-ha.yaml
```

Comment:
I see you use this approach in several places, is there any reason for not using the `--inplace` flag of `yq`? I think it would be better (1 command instead of 2 and with the `cp` approach you may have a typo in the destination filename that with in place you are sure is the same.

Reply:
I was not able to make it work with yq, maybe I did something wrong